src/mars.d

```diff
@@ -550,6 +548,47 @@ private int tryMain(size_t argc, const(char)** argv)
                 else
                     goto Lerror;
             }
+            else if (memcmp(p + 1, cast(char*)"mcpu", 4) == 0)
+            {
+                // Parse:
+                //      -mcpu=identifier
+                if (p[5] == '=')
+                {
+                    if (strcmp(p + 6, "?") == 0)
+                    {
+                        printf("
+CPU architectures supported by -mcpu=id:
+  =?             list information on all architecture choices
+  =baseline      use default architecture as determined by target
+  =avx           use AVX 1 instructions
+  =native        use CPU architecture that this compiler is running on
+");
+                        exit(EXIT_SUCCESS);
+                    }
+                    else if (Identifier.isValidIdentifier(p + 6))
```

Comment:
It's a copy/paste bit of code from the `"-transition=id"` block. Splitting out the parser would wind up being even more lines of code, since it's a one-off job. Of course, at some point we should just migrate the whole shebang to `std.getopt`, but I don't think it is worthwhile to badly reinvent `std.getopt` in the meantime.